src/endpoint.jl

```diff
@@ -82,26 +108,65 @@ function Base.isequal(a::RightEndpoint, b::LeftEndpoint)
 end
 
 function Base.isless(a::LeftEndpoint, b::LeftEndpoint)
-    a.endpoint < b.endpoint || (a.endpoint == b.endpoint && isclosed(a) && !isclosed(b))
+    return (
+        !isunbounded(b) && (
```

Comment:
I think using `isunbounded` consistently in these `Endpoint` comparison functions ends up being the most readable. Definitely is just a personal opinion though.

I should mention `isunbounded(::AbstractInterval)` and `isbounded(::AbstractInterval)` may surprise you since: `isunbounded(interval) != !isbounded(interval)`. This is due to `isunbounded` being defined as both endpoints are unbounded while `isbounded` is defined as both endpoints being bounded. This ends up being useful as `!isbounded(interval)` means at least one endpoint is unbounded.